resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/StreamingResourceServiceProxy.java

```diff
@@ -0,0 +1,176 @@
+
+    private final PersistenceLayer persistence;
+    private final ResourceService delegate;
+    private final ObjectMapper objectMapper;
+    private Optional<ResourceConfiguration> resourceConfiguration;
+
+    private final Serialiser<LeafResource> serialiser = new LineSerialiser<>() {
+        @Override
+        public String serialiseLine(final LeafResource obj) {
+            try {
+                return objectMapper.writeValueAsString(obj);
+            } catch (JsonProcessingException e) {
+                LOGGER.error("Encountered JSONProccessingException while serialising object {}", obj);
+                LOGGER.error("Exception was ", e);
+                throw new RuntimeException(e);
+            }
+        }
+
+        @Override
+        public LeafResource deserialiseLine(final String line) {
+            LOGGER.warn("No implementation of deserialiseLine, ignoring argument {}", line);
+            return null;
+        }
+    };
+
+    public StreamingResourceServiceProxy(final PersistenceLayer persistence, final ResourceService delegate, final ObjectMapper objectMapper) {
+        this.persistence = persistence;
+        this.delegate = delegate;
+        this.objectMapper = objectMapper;
+        this.resourceConfiguration = Optional.empty();
+    }
+
+    public StreamingResourceServiceProxy(final PersistenceLayer persistence, final ResourceService delegate, final ObjectMapper objectMapper, final ResourceConfiguration resourceConfiguration) {
+        this(persistence, delegate, objectMapper);
+        this.resourceConfiguration = Optional.of(resourceConfiguration);
+    }
+
+    @Transactional
+    public void getResourcesByResource(final Resource resource, final OutputStream outputStream) {
+        // Try first from persistence
+        LOGGER.info("Trying from persistence store");
+        Stream<LeafResource> resourceStream = persistence.getResourcesById(resource.getId())
+                // Otherwise call out to resource service
+                .orElseGet(() -> {
+                    LOGGER.info("Persistence empty, delegating to resource-service");
+                    // Persist returned resources as the stream is consumed
+                    return persistence.withPersistenceById(resource.getId(), delegate.getResourcesById(resource.getId()));
+                });
+        // Consume the stream, write to the output stream
+        serialiseAndWriteStreamToOutput(resourceStream, outputStream);
+    }
+
+    @Transactional
+    public void getResourcesById(final String resourceId, final OutputStream outputStream) {
+        // Try first from persistence
+        LOGGER.info("Trying from persistence store");
+        Stream<LeafResource> resourceStream = persistence.getResourcesById(resourceId)
+                // Otherwise call out to resource service
+                .orElseGet(() -> {
+                    LOGGER.info("Persistence empty, delegating to resource-service");

```

Comment:
These need to be at trace/debug level otherwise production logs get too noisy.

Reply:
My thinking here is that this is printed at most once per inbound request to the resource service, and gives an obvious sign of service vs persistence

Reply:
This is in-line with behaviour in policy and user services